### PR TITLE
Linked list

### PR DESCRIPTION
**Pull request by lhchandler4**

### CODE REVIEW

**Comment by lhchandler4** [comment]
- [ X] Top-level README “Table of Contents” is updated
- [ X] Feature tasks for this challenge are completed
- [ X] Unit tests written and passing
    - [ X] “Happy Path” - Expected outcome
    - [ X] Expected failure
    - [ X] Edge Case (if applicable/obvious)
- [ X] README for this challenge is complete
    - [ X] Summary, Description, Approach & Efficiency, Solution
    - [ X] Link to code

File: challenges/assets/README/LinkedList_README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Singly Linked List
+Linked Lists are a data structure that offer certain advantages over array lists
+
+## Challenge
+The challenge was to create three methods. Insert, Includes, Print
+
+## Approach & Efficiency
+Insert has a big O of 1. Includes has a big O of n. Print also has a big O of n.
```
Comment (evanslaton) [inline challenges/assets/README/LinkedList_README.md:8]: When talking about Big O notation you usually want to include both time and space. If they're the same then it's a good idea to mention it:

Insert has a Big O of 1 for both time and space

Also, the proper notation is O(1)

File: challenges/src/main/java/Java/LinkedList/LinkedList.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
+package Java.LinkedList;
+
+import java.util.*;
+
+public class LinkedList {
+    Node head;
```
Comment (evanslaton) [inline challenges/src/main/java/Java/LinkedList/LinkedList.java:6]: Definitely get in the habit specifying if instance variables are public, protected or private. We don't want to let Java decide anything if we don't have to

File: challenges/src/main/java/Java/LinkedList/LinkedList.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    public LinkedList() {
+        this.head = null;
+    }
+
+    public void append(int data) {
+        if (this.head == null) {
+            this.head = new Node(data);
+            return;
```
Comment (evanslaton) [inline challenges/src/main/java/Java/LinkedList/LinkedList.java:15]: If you wrap the rest of the code in this method in an `else` then you would't need to use return to break out of the method to stop the other code from running (your solution is valid too!!!!)

File: challenges/src/main/java/Java/LinkedList/LinkedList.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    public void insert(int data) {
+        Node newHead = new Node(data);
+        newHead.next = this.head;
+        this.head = newHead;
+    }
+
+    public boolean includes(int x) {
+        Node current = this.head;
+        while (current != null) {
+            if (current.data == x)
```
Comment (evanslaton) [inline challenges/src/main/java/Java/LinkedList/LinkedList.java:33]: While this will work, it's dangerous to have an `if` statement on multiple lines and not use `{}`

Without the `{}` only the very next line of code will be included in the `if`

File: challenges/src/test/java/Java/LinkedList/LinkedListTest.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        assertEquals(5, testNode.data);
+    }
+
+    @Test
+    public void test_Insert() {
+        LinkedList testing = new LinkedList();
+        testing.insert(6);
+        LinkedList secondTest = new LinkedList();
+        secondTest.insert(8);
+        assertEquals(6,testing.head.data);
```
Comment (evanslaton) [inline challenges/src/test/java/Java/LinkedList/LinkedListTest.java:20]: Since the 8 was inserted last, shouldn't `testing.head.data` be 8? Nevermind, didn't see that there were 2 LinkedLists

File: challenges/src/main/java/Java/LinkedList/LinkedList.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
+package Java.LinkedList;
+
+import java.util.*;
```
Comment (evanslaton) [inline challenges/src/main/java/Java/LinkedList/LinkedList.java:3]: This import statement isn't being used, be sure to delete unused import statements!
